File: Playlist/ToggleQueueTitleTransformer.m
//
//  ToggleQueueTitleTransformer.m
//  Cog
//
//  Created by Vincent Spader on 3/8/08.
//  Copyright 2008 __MyCompanyName__. All rights reserved.
//

#import "ToggleQueueTitleTransformer.h"

#import "Logging.h"

@implementation ToggleQueueTitleTransformer

+ (Class)transformedValueClass {
	return [NSNumber class];
}
+ (BOOL)allowsReverseTransformation {
	return NO;
}

// Convert from NSNumber to NSString
- (id)transformedValue:(id)value {
	DLog(@"VALUE: %@", value);
	if(value == nil) return nil;
	BOOL queued = [value boolValue];

	if(queued) {
		return @"Remove from Queue";
	} else {
		return @"Add to Queue";
	}
}

@end

File: Playlist/ToggleQueueTitleTransformer.swift
import Foundation

@objc(ToggleQueueTitleTransformer)
class ToggleQueueTitleTransformer: ValueTransformer {
    override class func transformedValueClass() -> AnyClass {
        return NSNumber.self
    }

    override class func allowsReverseTransformation() -> Bool {
        return false
    }

    // Convert from NSNumber to NSString
    override func transformedValue(_ value: Any?) -> Any? {
        if value == nil {
            return nil
        }

        let queued = (value as? NSNumber)?.boolValue ?? false

        if queued {
            return NSLocalizedString("RemoveFromQueue", comment: "Context menu item name for removing a track from the current queue")
        } else {
            return NSLocalizedString("AddToQueue", comment: "Context menu item name for adding a track to the current queue")
        }
    }
}
